torchtune/models/llama2/_model_builders.py

```diff
@@ -263,9 +264,91 @@ def lora_llama2_70b(
         quantize_base=quantize_base,
     )
 
+
 qlora_llama2_70b = partial(lora_llama2_70b, quantize_base=True)
 qlora_llama2_70b.__doc__ = """
 Builder for creating a Llama2 70B model with QLoRA enabled. Base model weights in linear layers
 that LoRA is applied to are quantized per the QLoRA paper: https://arxiv.org/abs/2305.14314.
 Please see `lora_llama2_70b` for full API arguments.
 """
+
+
+def llama2_reward_7b() -> TransformerDecoder:
+    """
+    Builder for creating a Llama2 model initialized w/ the default 7B parameter values
+    from https://arxiv.org/abs/2307.09288, where the output layer is a classification layer
+    projecting to a single class for reward modelling.
```

Comment:
There's actually a typo here, sorry, `llama2_classifier_7b` shouldn't exist. I think we're on the same page. The general pattern here is: `_classifier` exists in `_component_builders` as a general interfacew to a classifier model. The only use for these models currently in the codebase is for reward modelling for PPO, where reward models output a scalar (I do agree this constraint doesn't apply to all reward models). 

For simplicity, I could drop the `_reward` in model builders, and expose a `num_classes` in the model builder which is defaulted to 1. So you'd have:

```python
model_family/model_builders.py

... import model_family_classifier

def model_family_7b_classifier(num_classes: int = 1):
    return model_family_classifier(num_classes=num_classes, ...)

```

I think this would clean up the API, and offer flexibility to users who wish to define recipes for classification models  for which there is some interest from the community e.g. https://github.com/pytorch/torchtune/issues/1124.

How does this sound?

Reply:
I think the way you have it here where classifier is a generic component builder and reward is just num_classes = 1 in the model builders is great. We also don't want to add too many builder layers that add distance between what the user calls and the actual model architecture definition

Reply:
Thanks @RdoubleA. Let's roll with things as-is for now.